### PR TITLE
TVIST1-673: Added types to mail template custom fields

### PR DESCRIPTION
**Pull request by jekuaitk**
https://jira.itkdev.dk/browse/TVIST1-673

* Allows configuration of mail template custom fields to either text or textArea type.

### CODE REVIEW

File: src/Form/MailTemplateCustomDataType.php [hunk 1]
```diff
@@ -28,13 +29,23 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
...
 
-            foreach ($customFields as $key => $label) {
-                $builder->add($key, TextType::class, [
-                    'label' => $label,
-                    'mapped' => false,
-                    'data' => isset($options['data'][$key]) && !empty($options['data'][$key]) ? $options['data'][$key] : '',
-                    'required' => false,
-                ]);
+            foreach ($customFields as $key => $value) {
+                if ('textArea' === $value['type']) {
```
Comment (rimi-itk) [inline src/Form/MailTemplateCustomDataType.php:33]: Consider adding a `const`, `TYPE_TEXTAREA = 'textarea'`, say, in this class and use it in stead of writing `'textArea`' in multiple places.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,9 @@ about writing changes to this log.
 - [TVIST1-706](https://jira.itkdev.dk/browse/TVIST1-706):
   Added checkbox giving the option to not send receipt upon
   creating hearing post response
+- [TVIST1-673](https://jira.itkdev.dk/browse/TVIST1-673):
+  Allowed configuration of mail template custom fields to
+  either text or textArea.
```
Comment (rimi-itk) [inline CHANGELOG.md:20]: I would write “textarea” (all lowercase).

File: src/Form/MailTemplateCustomDataType.php [hunk 3]
```diff
@@ -28,13 +29,23 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
...
+                        'required' => false,
+                        'attr' => ['rows' => 5],
+                    ]);
+                } elseif ('text' === $value['type']) {
+                    $builder->add($key, TextType::class, [
+                        'label' => $value['label'],
+                        'mapped' => false,
+                        'data' => isset($options['data'][$key]) && !empty($options['data'][$key]) ? $options['data'][$key] : '',
+                        'required' => false,
+                    ]);
```
Comment (rimi-itk) [inline src/Form/MailTemplateCustomDataType.php:47]: Much of this code is the same Consider rewriting to something like

```php
$type  = TextType::class;
$options = [
  'label' => $value['label'],
  'mapped' => false,
  …
];

if ('textArea' === $value['type']) {
  $options['attr'] = …;
}

$builder->add($key, $type, $options);
```

Comment (jekuaitk) [reply src/Form/MailTemplateCustomDataType.php]: Updated in 81896df1e4a3900c171dacc9dbf863168bf41171.

File: CHANGELOG.md (hunk 2 above)
Comment (jekuaitk) [reply CHANGELOG.md]: Updated in 81896df1e4a3900c171dacc9dbf863168bf41171.

File: src/Form/MailTemplateCustomDataType.php (hunk 1 above)
Comment (jekuaitk) [reply src/Form/MailTemplateCustomDataType.php]: Added in 81896df1e4a3900c171dacc9dbf863168bf41171.

**Pull request merged**